packages/tables/frontend/sirius-components-tables/src/ForkViewMenuMenuAction.tsx

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2024 CEA LIST.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { gql, useMutation } from '@apollo/client';
+import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import Button from '@mui/material/Button';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {
+  CreateForkedStudioPayload,
+  CreateProjectSuccessPayload,
+  ForkViewMenuMenuActionProps,
+  GQLCreateForkedStudioInput,
+  GQLCreateForkedStudioMutationData,
+  GQLErrorPayload,
+} from './ForkViewMenuMenuAction.types';
+
+const forkViewMutation = gql`
```

Comment:
This action should be in a dedicated frontend module name something like `sirius-web-table` isolated from the table implementation. Only the sirius-web module should use it.